### ISSUE TITLE
Flaky test cmd/agent/app/reporter/grpc/TestBuilderWithCollectors/with_collector_connection_status_ready

### ISSUE THREAD

**Issue by yurishkuro** [jaegertracing/jaeger#2679]
https://github.com/jaegertracing/jaeger/pull/2657/checks?check_run_id=1486634709#step:5:346

```
=== RUN   TestBuilderWithCollectors/with_collector_connection_status_ready
    builder_test.go:182: 
        	Error Trace:	builder_test.go:182
        	Error:      	Not equal: 
        	            	expected: int(1)
        	            	actual  : int64(0)
        	Test:       	TestBuilderWithCollectors/with_collector_connection_status_ready
    builder_test.go:183: 
        	Error Trace:	builder_test.go:183
        	Error:      	Not equal: 
        	            	expected: int(1)
        	            	actual  : int64(0)
        	Test:       	TestBuilderWithCollectors/with_collector_connection_status_ready
    builder_test.go:184: 
        	Error Trace:	builder_test.go:184
        	Error:      	Not equal: 
        	            	expected: "127.0.0.1:41669"
        	            	actual  : ""
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-127.0.0.1:41669
        	            	+
        	Test:       	TestBuilderWithCollectors/with_collector_connection_status_ready
```

**Comment by yurishkuro** [comment]
Probably introduced in #2423 - cc @BernardTolosajr @jpkrohling

**Comment by yurishkuro** [comment]
But could've been dormant even earlier.

**Comment by BernardTolosajr** [comment]
I will check the issue. thanks

**Comment by Git-Jiro** [comment]
Hi!
How big of an issue is this flaky test?

I made some changes to the code and tested with:
> go test github.com/jaegertracing/jaeger/cmd/agent/app/reporter/grpc -run TestBuilderWithCollectors -count=10000 -failfast

What I saw so far:
The test case "TestBuilderWithCollectors/with_collector_connection_status_ready" is the only one where a real TCP connection is established. And it usually failed on my machine when the vast amount of test repetition caused some temporary resource issues in the systems TCP stack.
